tests/integration/shared-libs/cht-datasource/contact.spec.js

```diff
@@ -0,0 +1,381 @@
+        ' when limit and cursor is passed and cursor can be reused', async () => {
+        const freetext = 'contact';
+        const firstPage = await getUuidsPage({
+          ...Qualifier.byContactType(personType), ...Qualifier.byFreetext(freetext),
+        }, cursor, twoLimit);
+        const secondPage = await getUuidsPage({
+          ...Qualifier.byContactType(personType), ...Qualifier.byFreetext(freetext),
+        }, firstPage.cursor, twoLimit);
+        const expectedContactIds = [ contact0._id, contact1._id, contact2._id ];
+
+        const allData = [ ...firstPage.data, ...secondPage.data ];
+
+        expect(allData).excludingEvery([ '_rev', 'reported_date' ]).to.deep.equalInAnyOrder(expectedContactIds);
+        expect(firstPage.data.length).to.be.equal(2);
+        expect(secondPage.data.length).to.be.equal(1);
+        expect(firstPage.cursor).to.be.equal('2');
+        expect(secondPage.cursor).to.be.equal(null);
+      });
+
+      it('returns a page of place type contact ids' +
+        ' when limit and cursor is passed and cursor can be reused', async () => {
+        const firstPage = await getUuidsPage({
+          ...Qualifier.byContactType(placeType), ...Qualifier.byFreetext(placeFreetext),
+        }, cursor, twoLimit);
+        const secondPage = await getUuidsPage({
+          ...Qualifier.byContactType(placeType), ...Qualifier.byFreetext(placeFreetext),
+        }, firstPage.cursor, twoLimit);
+        const expectedContactIds = [ place0._id, clinic1._id, clinic2._id ];
+
+        const allData = [ ...firstPage.data, ...secondPage.data ];
+
+        expect(allData).excludingEvery([ '_rev', 'reported_date' ]).to.deep.equalInAnyOrder(expectedContactIds);
+        expect(firstPage.data.length).to.be.equal(2);
+        expect(secondPage.data.length).to.be.equal(1);
+        expect(firstPage.cursor).to.be.equal('2');
+        expect(secondPage.cursor).to.be.equal(null);
+      });
+
+      it('returns a page of unique contact ids for when multiple fields match the same freetext', async () => {
+        const expectedContactIds = [ contact0._id, contact1._id, contact2._id ];
+        const responsePage = await getUuidsPage(Qualifier.byFreetext(searchWord));
+        const responseIds = responsePage.data;
+        const responseCursor = responsePage.cursor;
+
+        expect(responseIds).excludingEvery([ '_rev', 'reported_date' ])
+          .to.deep.equalInAnyOrder(expectedContactIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of unique contact ids for when multiple fields match the same freetext with limit',
+        async () => {
+          const expectedContactIds = [ contact0._id, contact1._id, contact2._id ];
+          // NOTE: adding a limit of 4 to deliberately fetch 4 contacts with the given search word
+          // and enforce re-fetching logic
+          const responsePage = await getUuidsPage(Qualifier.byFreetext(searchWord), null, 4);
```

Comment:
Can you please use a limit that is lower than the amount of contacts that we expect to receive?